library/src/com/google/maps/android/data/kml/KmlRenderer.java

```diff
@@ -0,0 +1,524 @@
+     * ground overlays stored in the given containers
+     *
+     * @param groundOverlays ground overlays to add to the map
+     * @param kmlContainers  containers to check for ground overlays
+     */
+    private void addGroundOverlays(HashMap<KmlGroundOverlay, GroundOverlay> groundOverlays,
+            Iterable<KmlContainer> kmlContainers) {
+        addGroundOverlays(groundOverlays);
+        for (KmlContainer container : kmlContainers) {
+            addGroundOverlays(container.getGroundOverlayHashMap(),
+                    container.getContainers());
+        }
+    }
+
+    /**
+     * Adds all given ground overlays to the map
+     *
+     * @param groundOverlays hashmap of ground overlays to add to the map
+     */
+    private void addGroundOverlays(HashMap<KmlGroundOverlay, GroundOverlay> groundOverlays) {
+        for (KmlGroundOverlay groundOverlay : groundOverlays.keySet()) {
+            String groundOverlayUrl = groundOverlay.getImageUrl();
+            if (groundOverlayUrl != null && groundOverlay.getLatLngBox() != null) {
+                // Can't draw overlay if url and coordinates are missing
+                if (getImagesCache().get(groundOverlayUrl) != null) {
+                    addGroundOverlayToMap(groundOverlayUrl, mGroundOverlays, true);
+                } else if (!mGroundOverlayUrls.contains(groundOverlayUrl)) {
+                    mGroundOverlayUrls.add(groundOverlayUrl);
+                }
+            }
+        }
+    }
+
+    /**
+     * Downloads images of all ground overlays
+     */
+    private void downloadGroundOverlays() {
+        mGroundOverlayImagesDownloaded = true;
+        for (Iterator<String> iterator = mGroundOverlayUrls.iterator(); iterator.hasNext(); ) {
+            String groundOverlayUrl = iterator.next();
+            new GroundOverlayImageDownload(groundOverlayUrl).execute();
+            iterator.remove();
+        }
+    }
+
+    /**
+     * Adds ground overlays from a given URL onto the map
+     *
+     * @param groundOverlayUrl url of ground overlay
+     * @param groundOverlays   hashmap of ground overlays to add to the map
+     */
+    private void addGroundOverlayToMap(String groundOverlayUrl,
+            HashMap<KmlGroundOverlay, GroundOverlay> groundOverlays, boolean containerVisibility) {
+        BitmapDescriptor groundOverlayBitmap = BitmapDescriptorFactory
+                .fromBitmap(getImagesCache().get(groundOverlayUrl));
+        for (KmlGroundOverlay kmlGroundOverlay : groundOverlays.keySet()) {
+            if (kmlGroundOverlay.getImageUrl().equals(groundOverlayUrl)) {
+                GroundOverlayOptions groundOverlayOptions = kmlGroundOverlay.getGroundOverlayOptions()
+                        .image(groundOverlayBitmap);
+                GroundOverlay mapGroundOverlay = attachGroundOverlay(groundOverlayOptions);
+                if (containerVisibility == false) {
```

Comment:
Nit - for code style, to match other places this should be `!containerVisibility`.